### PR TITLE
Decorate class review

### PR DESCRIPTION
**Pull request by akezeth**
- Created Nameable class and used it as a parent for Person class.
- Implemented correct_name method within Person class.
- Created BaseDecorator class.
- Created CapitalizeDecorator class.
- Created TrimmerDecorator class.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-require_relative person
+require_relative 'person'
```
Comment (Shedrack-Sunday) [inline student.rb:2]: - [ ] One of the requirements here is that when setting the `classroom` for a `student` it also adds it to the classrooms' students: Which means you need to set a getter for the classroom class which you did not create initially. Kindly add the `getter 👍 

### Instructions:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hi @akezeth 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 🔴 

You have submitted a milestone of `doctorate-class` for `OOP school library: set up associations`. Kindly send the PR to the appropriate link

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by Nemwel-Boniface — COMMENTED** [review]
Hello @akezeth :wave: 

I would like to apologize to you on behalf of the last CR @Shedrack-Sunday . You seem to have submitted this PR for review in the wrong milestone which is "OOP school library: set up associations" which is why the CR seems to have requested changes for the missing requirements from that milestone.

The CR however was supposed to invalidate the project because you submitted it on the wrong milestone and that would have allowed you to be able to now submit in the correct milestone which is "OOP school library: decorate a class".

**What needs to happen next:**

1. No action is required from you I would only caution you to be more keen when submitting your projects for review. You will soon receive a review for the correct milestone.
2. As the previous reviewer @Shedrack-Sunday made an erroneous review for the "OOP school library: set up associations" and now in your dashboard it states "requires changes" for that milestone, it will soon be invalidated and you will be able to request for a review as first review when you are ready for that milestone to be reviewed.

**Note:** the "OOP school library: set up associations" milestone will be invalidated in a few hours so do not worry much. By thetime you get to submit it for review the review will have been reversed.

I hope this clears your doubts and also puts you at ease.

Incase anything is not clear please tag me @nemwel-boniface so that I get notified.

Till then #happycoding :rocket:

**Review by KanzaTahreem — APPROVED** [review]
# Approved 🙌

Hi @akezeth 

Great work nailing it on the first attempt 💪 **_Your project is COMPLETE! There is nothing else to say other than... it's time to merge it! 🚀 Congratulations!_**



## Highlights 🎯
There are some excellent points that you have made perfectly. Keep up the great work! 👏

-  ✅ Class is decorated, perfect!
-  ✅ Great work using the right GitFlow 
-  ✅ Linters are working properly, great job! 

## Optional suggestions ❕
N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**